### PR TITLE
Finish migration to hugo

### PR DESCRIPTION
**Pull request by seproDev**
I tried my hand at finally finishing this. For some things like menu structure I was a bit more liberal at making changes. Maybe the docs should be moved to a 3.4 section, while 3.2 is just the HTML of the old site like 3.0.

The site can be previewed here: https://seprodev.github.io/AegiSite/

### CODE REVIEW

**Comment by seproDev** [comment]
A couple last things to note:
If we want to maintain compatibility with old links:
 - `docs.aegisub.org/*` needs to redirect to `aegisub.org/docs/*` (assuming the site will be at `aegisub.org` [this also still needs to be reflected in the baseurl])
 - Either upercase links need to redirect to lowercase ones or we need to add `disablePathToLower: true` to the config
 
If we wanted to we could also try and maintain compatibility with the old blog URLs. Not sure if it is worth it though.
 
Should a blog post be written about the new website? Since updating content will become way easier, encouraging PRs seems like a good idea. Maybe a bit about the planned future of Aegisub can also be written. Alternatively depending on how far away it is the updated website could also just be mentioned in the 3.4.0 release post.

**Comment by inkydragon** [comment]
LGTM. 

@seproDev Just one question: "Migrate Chinese site part 1/2" You merged these changes from [Chinese-aegisite](https://github.com/computerfan/Chinese-aegisite), right?
If so, we need to clearly point out their (translators') contribution.

----

> A couple last things to note

There is always plenty of improvement we can do.
You can track these issues by opening a new issue at https://github.com/Aegisub/aegisite/issues.

But personally, I think it's a priority to merge the existing improvements into the main branch, and then let everyone contribute on the main repo.

**Comment by seproDev** [comment]
> [Chinese-aegisite](https://github.com/computerfan/Chinese-aegisite)

Oh I didn't find that repo when I was looking for it so instead I parsed the HTML data on the site. Now with the source files I will go through and see if there are any differences I missed.
And yes, contributions should be mentioned. Maybe adding it to the footer?

> I think it's a priority to merge the existing improvements into the main branch, and then let everyone contribute on the main repo.

Fair point. I'll create issues there once your branch is merged.

File: config/_default/params/params.zh-cn.yaml [hunk 1]
```diff
@@ -48,31 +65,48 @@ portraitPhotoWidths:
...
 
-# footer/left side
+# TODO: Maybe mention Hosting? NetworkRedux? Github Pages? Cloudflare Pages?
+# Footer
 footer: >-
-  Powered by <a href="https://www.networkredux.com/">NetworkRedux</a>, <a
-  href="https://gohugo.io/">Hugo</a>, and <a
-  href="https://getdoks.org/">Doks</a>
+  Powered by <a class="text-muted" href="https://gohugo.io/">Hugo</a> and <a
+  class="text-muted" href="https://getdoks.org/">Doks</a>
```
Comment (inkydragon) [inline config/_default/params/params.zh-cn.yaml:74]:
```suggestion
  class="text-muted" href="https://getdoks.org/">Doks</a>. <br/>
  Copyright (c) AegiSite <a href="https://github.com/Aegisub/aegisite/graphs/contributors">contributors</a>
    and <a href="https://github.com/computerfan/Chinese-aegisite/graphs/contributors">translators</a>.
```
image

Comment (seproDev) [reply config/_default/params/params.zh-cn.yaml:74]: I like it. I'll add the same text without the "and translators" to the English version.

**Pull request merged**